dbt_cloud_plugin/operators/dbt_cloud_check_model_result_operator.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+import json
+import requests
+import time
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import AirflowException, AirflowSkipException
+from ..hooks.dbt_cloud_hook import DbtCloudHook
+
+
+SUCCESSFUL_STATUSES = ['success', 'pass']
+
+
+class DbtModelException(Exception):
+    pass
+
+
+class DbtModelFailedException(DbtModelException):
+    pass
+
+
+class DbtModelNotRunException(DbtModelException):
+    pass
+
+
+class DbtCloudCheckModelResultOperator(BaseOperator):
+
+    template_fields = ['dbt_cloud_run_id']
+
+    @apply_defaults
+    def __init__(self, dbt_cloud_conn_id=None, dbt_cloud_run_id=None, model_names=None, ensure_models_ran=True, *args, **kwargs):
+        super(DbtCloudCheckModelResultOperator, self).__init__(*args, **kwargs)
+
+        if dbt_cloud_conn_id is None:
```

Comment:
If these params are required, why not just not assign default values?

Reply:
There's not really a sensible default value. Well, ok, for this conn_id in particular, it could default to `dbt_default`, which is the general pattern I've seen from other operators. But the other 2 required params really don't have any defaults; you need to specify them yourself.

Reply:
This comment was mostly curiosity as I've seen this structure in other operators, where the default value is None and then you have a series of if statements to check if something was input, as opposed to not setting a default and just letting the constructor throw an error if the param is missing. I was thinking like if you just had line 32 say: `def __init__(self, dbt_cloud_conn_id, dbt_cloud_run_id, model_names, ensure_models_ran=True, *args, **kwargs):`